packages/adp-tooling/src/types.ts

```diff
@@ -138,3 +142,119 @@ export const enum HttpStatusCodes {
     NOT_IMPLEMETED = 501,
     SERVICE_UNAVAILABLE = 503
 }
+
+export type Writer = new (fs: Editor, projectPath: string) => IWriter<any>;
+export type IWriterData<T extends ChangeType> = IWriter<GeneratorData<T>>;
+export interface IWriter<T> {
+    /**
+     * Writer function.
+     *
+     * @param {T} data - Data needed for the writer function.
+     */
+    write(data: T): Promise<void>;
+}
+
+export const enum ChangeType {
```

Comment:
Please add a concise description to each new type, since not all of them are self explanatory.

Reply:
Changed in: [706928d](https://github.com/SAP/open-ux-tools/pull/1643/commits/706928d0e145dee24ff454510c00cb9832238bec)